vars/registerDns.groovy

```diff
@@ -0,0 +1,53 @@
+import uk.gov.hmcts.contino.Consul
+import uk.gov.hmcts.contino.AppPipelineConfig
+
+def call(Map params) {
+  AppPipelineConfig config = params.appPipelineConfig
+
+  withAksClient(params.subscription, params.environment) {
+    Consul consul = new Consul(this, params.environment)
+
+    // Staging DNS registration
+    if (config.legacyDeployment) {
+      withIlbIp(params.environment) {
+        consul.registerDns("${params.product}-${params.component}-${params.environment}-staging", env.TF_VAR_ilbIp)
+      }
+    }
+
+    aksSubscriptionName = params.aksSubscription != null ? params.aksSubscription.name : null
+
+    // Note: update this when we get a PROD subscription
+    if (config.aksStagingDeployment) {
+      if (aksSubscriptionName && !aksSubscriptionName.contains('PROD')) {
+        def ingressIP = params.aksSubscription.ingressIp()
+        consul.registerDns("${params.product}-${params.component}-staging", ingressIP)
+      } else {
+        echo "Skipping dns registration for AKS as this environment is not configured with it: ${aksSubscriptionName}"
+      }
+    }
+    // AAT + PROD DNS registration
+    def aksEnv = params.aksSubscription != null && params.aksSubscription.envName
+
+    if (config.legacyDeployment) {
+      withIlbIp(params.environment) {
+        consul.registerDns("${params.product}-${params.component}-${params.environment}", env.TF_VAR_ilbIp)
```

Comment:
Can this be moved to line 14 (after the `-staging` registration)? Save the cal to reasonably expensive `withIlbIp`